crates/core/src/contracts/account.rs

```diff
@@ -39,7 +39,10 @@ pub trait Account<'a, P: Provider + Send + Sync + 'a> {
 
         // Make the function call to get the Starknet contract address
         let bytecode = self.provider().call(request, block_id).await.or_else(|err| match err {
-            ProviderError::StarknetError(StarknetError::ContractNotFound) => Ok(vec![]),
+            ProviderError::StarknetError(StarknetErrorWithMessage {
+                code: MaybeUnknownErrorCode::Known(StarknetError::ContractNotFound),
+                ..
```

Comment:
what is the difference between .. here and above writing an empty string message

Reply:
so this is needed because the `ProviderError::StarknetError` enum changed, meaning in now need to match on a `StarknetErrorWithMessage` structure.